services/collection/collection.cpp

```diff
@@ -48,8 +48,18 @@ namespace services::collection {
     auto collection_t::size(session_id_t& session) -> void {
         trace(log(), "collection {}::size", name_.collection);
         auto dispatcher = current_message()->sender();
-        auto result = dropped_ ? result_size() : result_size(size_());
-        actor_zeta::send(dispatcher, address(), handler_id(route::size_finish), session, result);
+        if (dropped_) {
+            actor_zeta::send(dispatcher, address(), handler_id(route::size_finish), session,
+                             make_cursor(context_->resource(), error_code_t::collection_dropped));
+        } else {
+            auto* sub_cursor = new sub_cursor_t(context_->resource(), address());
```

Comment:
@kotbegemot This place(how we add docs into cursor) looks like can be improvement in the future. For now we can leave it as it is.